impress-org/givewp
# Feature: improve and consolidate amount messages

## Description

This PR is a minor messaging tweak mixed with some DRY refactoring. The PMs decided that "repeat" is better than "occurs" and "donations" is better than "payments" in the donor-facing messages. The logic for creating the messages existed both in the donation form and the form builder, so this was consolidated as well.

Note that this affects both fixed (one-time) donation amount and recurring amount messages.

## Affects

The amount message displayed to the donor in the form builder and form.

## Visuals

<img width="745" alt="image" src="https://github.com/impress-org/givewp/assets/2024145/325e4dd3-987b-4a38-aeea-4c7c6f185931">

<img width="695" alt="image" src="https://github.com/impress-org/givewp/assets/2024145/cc57080c-3543-4eba-b3f7-f2e3a9f8acf2">

## Testing Instructions

1. Make a v3 form
2. Give it a fixed amount
3. Check the message in the builder and front-end
4. Edit the form and add recurring with (and without) various intervals and installments
5. Check the front-end again
6. Mess around with said settings more and keep checking front-end

## Pre-review Checklist
